tiledb/sm/query/readers/reader_base.cc

```diff
@@ -1576,10 +1640,24 @@ Status ReaderBase::unfilter_tiles(
           // Unfilter 't' for fixed-sized tiles, otherwise unfilter both 't' and
           // 't_var' for var-sized tiles.
           if (!var_size) {
-            if (!nullable)
-              RETURN_NOT_OK(unfilter_tile(name, t));
-            else
+            if (!nullable) {
+              if (array_schema_.bitsort_filter_attr().has_value()) {
+                std::vector<Tile*> dim_tiles;
```

Comment:
I've been thinking and there might be some extra cleanup we could do here... Is there any way that dim_tiles, qb_vector and db live in bitsort_metadata? That way, construct_bitsort_filter_argument could create everything and the calling function doesn't need to declare any variables to keep alive.

Reply:
I can't turn either QueryBuffer or DomainBuffersView into incomplete classes, since they rely on dimension, and dimension relies on filter pipeline, which would create that circular dependency. I could pass them in as pointers, but I'd still need space to store them. I could either allocate memory to do so, or just declare them as I did before. The only thing I think I could store is dim_tiles.